### PR TITLE
Expose javadocs via the IFabricResultSaver

### PR DESCRIPTION
**Pull request by modmuss50**

### CODE REVIEW

File: src/net/fabricmc/fernflower/api/IFabricResultSaver.java [hunk 1]
```diff
@@ -2,7 +2,22 @@
...
 
 import org.jetbrains.java.decompiler.main.extern.IResultSaver;
+import org.jetbrains.java.decompiler.struct.StructClass;
+import org.jetbrains.java.decompiler.struct.StructField;
+import org.jetbrains.java.decompiler.struct.StructMethod;
 
 public interface IFabricResultSaver extends IResultSaver {
   void saveClassEntry(String path, String archiveName, String qualifiedName, String entryName, String content, int[] mapping);
+
+  default String getClassDoc(StructClass structClass) {
```
Comment (natanfudge) [inline src/net/fabricmc/fernflower/api/IFabricResultSaver.java]: Put `@Nullable` if we have it

File: src/org/jetbrains/java/decompiler/main/ClassWriter.java [hunk 2]
```diff
@@ -31,9 +33,15 @@
 
 public class ClassWriter {
   private final PoolInterceptor interceptor;
+  private final StructContext context;
```
Comment (natanfudge) [inline src/org/jetbrains/java/decompiler/main/ClassWriter.java]: `@Nullable` here

File: src/org/jetbrains/java/decompiler/main/ClassWriter.java [hunk 3]
```diff
@@ -943,6 +957,15 @@ private static void appendComment(TextBuffer buffer, String comment, int indent)
     buffer.appendIndent(indent).append("// $FF: ").append(comment).appendLineSeparator();
   }
 
+  private static void appendJavaDoc(TextBuffer buffer, String javaDoc, int indent) {
+    if(javaDoc == null) return;
```
Comment (natanfudge) [inline src/org/jetbrains/java/decompiler/main/ClassWriter.java]: formatting is wrong here

**Review by natanfudge — APPROVED** [review]

File: src/net/fabricmc/fernflower/api/IFabricResultSaver.java (hunk 1 above)
Comment (modmuss50) [reply src/net/fabricmc/fernflower/api/IFabricResultSaver.java]: Dont have it

File: src/org/jetbrains/java/decompiler/main/ClassWriter.java (hunk 3 above)
Comment (modmuss50) [reply src/org/jetbrains/java/decompiler/main/ClassWriter.java]: fixed.

File: src/net/fabricmc/fernflower/api/IFabricResultSaver.java (hunk 1 above)
Comment (natanfudge) [reply src/net/fabricmc/fernflower/api/IFabricResultSaver.java]: Really? This is Jetbrains software.

**Review by Player3324 — APPROVED** [review]

**Review by asiekierka — APPROVED** [review]

**Review by ChloeDawn — APPROVED** [review]

**Pull request merged**